src/main/java/com/exactpro/th2/healer/cache/EventsCache.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2020-2021 Exactpro (Exactpro Systems Limited)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.exactpro.th2.healer.cache;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReadWriteLock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+
+public class EventsCache<K, V> extends LinkedHashMap<K, V> {
+    private final int capacity;
+    private final Lock writeLock;
+    private final Lock readLock;
+
+    public EventsCache(int capacity) {
+        super(capacity);
```

Comment:
Are you sure that we need to have an initial capacity equals to the max capacity of the cache?